R/clean_data_helpers.R

```diff
@@ -65,12 +38,185 @@ scan_columns <- function(x) {
 #'   data = readRDS(system.file("extdata", "messy_data.RDS",
 #'                              package = "cleanepi"))
 #' )
+#'
+#' @details
+#' For columns of type character, the detected numeric values could actually be
+#' of type Date or date-time. This is because R coerces some Date values into
+#' numeric when the date is imported from an MS Excel file.
+#'
 scan_data <- function(data) {
-  scan_result           <- data.frame(t(apply(data, 2L, scan_columns)))
-  names(scan_result)    <- c("missing", "numeric", "date", "character",
-                             "logical")
-  row_names             <- rownames(scan_result)
-  rownames(scan_result) <- NULL
-  scan_result           <- cbind(Field_names = row_names, scan_result)
+  types       <- vapply(data, typeof, character(1L))
+  scan_result <- vapply(seq_len(ncol(data)), function(col_index) {
+    scan_columns(data[[col_index]], types[[col_index]])
+  }, numeric(7L))
+  scan_result        <- as.data.frame(t(scan_result))
+  names(scan_result) <- c("missing", "numeric", "date", "character",
+                          "logical", "date-time", "factor")
```

Comment:
Why did this change? It doesn't give the impression that the list results from an explicit and motivated design choice?

Reply:
It seems that this function is now mostly returning the `class()`. Can we go back to the drawing board and write a list of requirements:

- when I have X type of data, I want to know Y to inform Z

> For example, a column with both numeric and character values might require a call for convert_to_numeric(), or a conversion of the numeric values into character, or setting the values of one of these type to NA. With this information in hand, the user can plan the set of cleaning operations to apply on the dataset.

This is a good example for analyzing types in `character` columns. What about the others?